fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1748,32 +1745,63 @@ public Builder setLeaseLengthMillis(long leaseLengthMillis) {
         }
 
         /**
-         * Add an {@link IndexingPolicy} policy. If set, this policy will be used to build the target index by only
-         * scanning records of a source index, avoiding  a full record scan.
+         * This method will soon be deprecated, please prefer {@link #setInexingPolicyBuilder}.
+         * Add an {@link IndexingPolicy} policy. If set, this policy will determine how the index should be
+         * built.
          * @param indexingPolicy see {@link IndexingPolicy}
          * @return this Builder
          */
         public Builder setIndexingPolicy(@Nullable final IndexingPolicy indexingPolicy) {
+            this.indexingPolicyBuilder = null;
             if (indexingPolicy == null) {
-                this.indexingPolicy = IndexingPolicy.defaultPolicy();
+                this.indexingPolicy = IndexingPolicy.DEFAULT;
             } else {
                 this.indexingPolicy = indexingPolicy;
             }
             return this;
         }
 
+        /**
+         * Add an {@link IndexingPolicy.Builder}. If set, this policy will determine how the index should be
+         * built.
+         * @param builder an IndexingPolicy builder.
+         * @return this Builder
+         */
+        public Builder setInexingPolicyBuilder(@Nonnull final IndexingPolicy.Builder builder) {
+            this.indexingPolicy = null;
+            this.indexingPolicyBuilder = builder;
+            return this;
+        }
+
         /**
          * Build an {@link OnlineIndexer}.
          * @return a new online indexer
          */
         public OnlineIndexer build() {
             validate();
-            if (explicitIndexStatePrecondition) {
-                indexingPolicy.ifDisabled = indexStatePrecondition.ifDisabled;
-                indexingPolicy.ifWriteOnly = indexStatePrecondition.ifWriteOnly;
-                indexingPolicy.ifMismatchPrev = indexStatePrecondition.ifMismatchPrev;
-                indexingPolicy.ifReadable = indexStatePrecondition.ifReadable;
+            // Here: Supporting dual backward compatibilities:
+            // 1. Allow the use of indexStatePrecondition
+            // 2. Allow the use of an explicit index policy (instead of the index policy builder)
```

Comment:
Okay, and if I'm understanding the semantics here, they are:

1. If you set both an `IndexingPolicy` and an `IndexingPolicy.Builder` on the builder, it will end up using whichever you set last between the two of them
2. If you set an `IndexStatePrecondition` and an `IndexingPolicy` (or builder), then the values of the `DesiredAction`s will all be set from the `IndexStatePrecondition`, regardless of what's in the policy.

I guess those are reaonsable enough semantics, and probably backwards compatible (i.e., those who don't realize they need to set the `IndexingPolicy`'s `DesiredActions` will see it "do the right thing). But it's probably worth documenting in the corresponding `.setX` method Javadocs

Reply:
1. Yes. I was trying to imitate the use of last value in other builder function (i.e `builder.setX('a').setX('b')` sets 'b' in X)
2. Yes. As a backward compatibility for today's code.
I'll try to improve the documentation.